### PR TITLE
Updates JDT and fix a casting bug from JDT

### LINKED ISSUE #328: Casting bug with generic invocation
When you try to apply Spoon on this sample:

```java
public class Castings {
	public void test(double a) {}

	public void foo() {
		List<Integer> list = new ArrayList<Integer>(1);
		list.add(1);
		test(getValue(list));
	}

	public final <T> T getValue(List<T> list)
	{
		return list.get(0);
	}
}
```

JDT isn't happy and throws this error:

```console
Type mismatch: cannot convert from Integer to double at D:\Programmieren\CubeEngine\spoon\src\test\java\spoon\test\casts\Castings.java:12
```

You can read your [issue on the bugtracker of JDT](https://bugs.eclipse.org/bugs/show_bug.cgi?id=464496#c2) and you can have the history of this issue on this [PR](https://github.com/INRIA/spoon/pull/292).

### PR DESCRIPTION
**Pull request by GerardPaligot**
Closes #328

### CODE REVIEW

**Comment by monperrus** [comment]
add API doc in CtInvocation.getType:

Return the type returned by the invocation. If the invocation is to a method where the returned type is a generic type, this method returns the actual type bound to this particular invocation.

File: src/test/java/spoon/test/casts/CastTest.java [hunk 1]
```diff
@@ -65,4 +69,21 @@ public void testCast3() {
...
+			@Override
+			public boolean matches(CtInvocation element) {
+				return "getValue".equals(element.getExecutable().getSimpleName()) && super.matches(element);
+			}
+		}).get(0);
+
+		assertEquals("T", getValueMethod.getType().getSimpleName());
+		assertEquals("T", getValueMethod.getParameters().get(0).getType().getActualTypeArguments().get(0).toString());
+
+		assertEquals(type.getFactory().Class().INTEGER, getValueInvocation.getType());
```
Comment (monperrus) [inline src/test/java/spoon/test/casts/CastTest.java]: contract:  If the invocation is to a method where the returned type is a generic type, getType returns the actual type bound to this particular invocation.

Comment (GerardPaligot) [inline src/test/java/spoon/test/casts/CastTest.java]: Done.

**Comment by GerardPaligot** [comment]
@monperrus Travis ok.

**Pull request merged**